checks/raw/pinned_dependencies_test.go

```diff
@@ -2084,3 +2084,212 @@ func TestCollectGitHubActionsWorkflowPinning(t *testing.T) {
 		})
 	}
 }
+
+func TestCsProjAnalysis(t *testing.T) {
+	t.Parallel()
+
+	//nolint:govet
+	tests := []struct {
+		unlocked        int
+		expectErrorLogs bool
+		name            string
+		filename        string
+	}{
+		{
+			name:            "empty file",
+			filename:        "./testdata/dotnet-empty.csproj",
+			expectErrorLogs: true,
+		},
+		{
+			name:     "locked mode enabled",
+			filename: "./testdata/dotnet-locked-mode-enabled.csproj",
+			unlocked: 0,
+		},
+		{
+			name:     "locked mode disabled",
+			filename: "./testdata/dotnet-locked-mode-disabled.csproj",
+			unlocked: 1,
+		},
+		{
+			name:     "locked mode disabled implicitly",
+			filename: "./testdata/dotnet-locked-mode-disabled-implicitly.csproj",
+			unlocked: 1,
+		},
+		{
+			name:            "invalid file",
+			filename:        "./testdata/dotnet-invalid.csproj",
+			expectErrorLogs: true,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt // Re-initializing variable so it is not changed while executing the closure below
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			var content []byte
+			var err error
+
+			content, err = os.ReadFile(tt.filename)
+			if err != nil {
+				t.Errorf("cannot read file: %v", err)
```

Comment:
normally in cases where the test case can't continue, you'd use `t.Fatalf` instead of `t.Errorf`.